apps/meteor/app/livechat/server/hooks/afterUserActions.ts

```diff
@@ -1,23 +1,50 @@
 import type { IUser } from '@rocket.chat/core-typings';
-import { Users } from '@rocket.chat/models';
-import type { UsersUpdateParamsPOST } from '@rocket.chat/rest-typings';
 
 import { callbacks } from '../../../../lib/callbacks';
+import { Livechat } from '../lib/Livechat';
+import { callbackLogger } from '../lib/logger';
 
-type UserData = UsersUpdateParamsPOST['data'] & { _id: string };
+type IAfterSaveUserProps = {
+	user: IUser;
+	oldUser: IUser | null;
+};
+
+const wasAgent = (user: Pick<IUser, 'roles'> | null) => user?.roles?.includes('livechat-agent');
+const isAgent = (user: Pick<IUser, 'roles'> | null) => user?.roles?.includes('livechat-agent');
+
+const handleAgentUpdated = async (userData: IAfterSaveUserProps) => {
+	const {
+		user: { _id: userId, username },
+		oldUser,
+	} = userData;
+
+	if (wasAgent(oldUser) && !isAgent(userData.user)) {
+		callbackLogger.debug('Removing agent', userId);
+		await Livechat.removeAgent(username);
+	}
 
-const handleAgentUpdated = async (userData: UserData) => {
-	if (!userData?.roles?.includes('livechat-agent')) {
-		await Users.unsetExtension(userData._id);
+	if (!wasAgent(oldUser) && isAgent(userData.user)) {
```

Comment:
if you're destructuring oldUser, you may destructure `user` as well :)

Reply:
You're right. That's much better! Thanks :hugs: